src/stdlib/io/file.rs

```diff
@@ -0,0 +1,323 @@
+
+            loop {
+                match seek_read(file, position, &mut byte) {
+                    Ok(0) => {
+                        if buf.is_empty() {
+                            return Ok(None);
+                        }
+                        break;
+                    }
+                    Ok(_) => {
+                        if byte[0] == b'\n' {
+                            break;
+                        } else if byte[0] == b'\r' {
+                            let mut next_byte = [0u8; 1];
+                            match seek_read(file, position, &mut next_byte) {
+                                Ok(1) => {
+                                    if next_byte[0] != b'\n' {
+                                        // Wasn't \r\n, but we can't put it back if the stream doesn't support seek
+                                        // We'll just consider this a part of the next line when reading continues
+                                    }
+                                }
+                                _ => {}
+                            }
+                            break;
+                        } else {
+                            buf.push(byte[0]);
+                        }
+                    }
+                    Err(err) => return Err(err.to_string().into_value(ctx).into()),
+                }
+            }
+
+            Ok(Some(ctx.intern(&buf).into()))
+        }
+
+        "L" => {
+            let mut buf = Vec::new();
+            let mut byte = [0u8; 1];
+
+            loop {
+                match seek_read(file, position, &mut byte) {
+                    Ok(0) => {
+                        if buf.is_empty() {
+                            return Ok(None);
+                        }
+                        break;
+                    }
+                    Ok(_) => {
+                        buf.push(byte[0]);
+                        if byte[0] == b'\n' {
+                            break;
+                        } else if byte[0] == b'\r' {
+                            let mut next_byte = [0u8; 1];
+                            match seek_read(file, position, &mut next_byte) {
+                                Ok(1) => {
+                                    if next_byte[0] == b'\n' {
+                                        *position += 1;
+                                        buf.push(next_byte[0]);
+                                    } else {
+                                        // Again, we can't put back what we read in a generic Read
+                                        // This byte will be part of the next line
+                                    }
+                                    break;
+                                }
+                                _ => break,
+                            }
+                        }
+                    }
+                    Err(err) => return Err(err.to_string().into_value(ctx).into()),
+                }
+            }
```

Comment:
Lua's readline only treats `\n` as newlines, so this can just read until it finds a `\n` byte.  I'd recommend using `BufRead::read_until` here instead, as it can handle it more efficiently than making a read for each byte.